pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java

```diff
@@ -15,224 +15,101 @@
-      ChannelFuture f = ctx.close();
-      f.addListener(this);
-      LOGGER.debug("Channel close future added !!");
-    }
-
-    @Override
-    public void operationComplete(ChannelFuture future) throws Exception {
-      LOGGER.debug("Channel is closed !!");
-    }
+  @Test
+  public void testCloseServerChannel()
+      throws Exception {
+    Assert.assertTrue(_nettyTCPClientConnection.connect());
+    ResponseFuture responseFuture =
+        _nettyTCPClientConnection.sendRequest(Unpooled.wrappedBuffer(NettyTestUtils.DUMMY_REQUEST.getBytes()), 1L,
+            5000L);
+
+    closeServerConnection();
+
+    _countDownLatch.countDown();
+    ByteBuf serverResponse = responseFuture.getOne();
+    Assert.assertNull(serverResponse);
+    Assert.assertFalse(responseFuture.isCancelled());
+    Assert.assertNotNull(responseFuture.getError());
   }
 
-  /**
-   * Server Channel Initializer
-   */
-  protected static class MyServerChannelInitializer extends ServerChannelInitializer {
-
-    public MyServerChannelInitializer(RequestHandlerFactory handlerFactory) {
-      super(handlerFactory, null, null);
-    }
-
-    @Override
-    protected void initChannel(SocketChannel ch) throws Exception {
-      super.initChannel(ch);
-      ch.pipeline().removeLast();
-      ch.pipeline().addLast("request_handler", new MyNettyChannelInboundHandler());
-      LOGGER.debug("Server Channel pipeline setup modified. Pipeline: {}", ch.pipeline().names());
-
-    }
+  @AfterMethod
+  public void tearDown()
+      throws Exception {
+    closeClientConnection();
+    closeServerConnection();
   }
 
-  private static class MyRequestHandler implements RequestHandler {
-    private String _response;
-    private String _request;
-    private final CountDownLatch _responseHandlingLatch;
-
-    public MyRequestHandler(String response, CountDownLatch responseHandlingLatch) {
-      _response = response;
-      _responseHandlingLatch = responseHandlingLatch;
-    }
-
-    @Override
-    public ListenableFuture<byte[]> processRequest(ChannelHandlerContext channelHandlerContext, ByteBuf request) {
-      byte[] b = new byte[request.readableBytes()];
-      request.readBytes(b);
-      if (null != _responseHandlingLatch) {
-        try {
-          _responseHandlingLatch.await();
-        } catch (InterruptedException e) {
-          e.printStackTrace();
-        }
-      }
-      _request = new String(b);
-
-      //LOG.info("Server got the request (" + _request + ")");
-      return Futures.immediateFuture(_response.getBytes());
-    }
-
-    public String getRequest() {
-      return _request;
-    }
-
-    public String getResponse() {
-      return _response;
-    }
-
-    public void setResponse(String response) {
-      _response = response;
-    }
+  private void closeClientConnection()
+      throws Exception {
+    _nettyTCPClientConnection.close();
+    Uninterruptibles.sleepUninterruptibly(100L, TimeUnit.MILLISECONDS);
```

Comment:
please add a comment here as to why the sleep is needed (i.e. what event are you waiting for)